resolver/caching_resolver.go

```diff
@@ -111,21 +111,33 @@ func (r *CachingResolver) reloadCacheEntry(ctx context.Context, cacheKey string)
 	logger.Debugf("prefetching '%s' (%s)", util.Obfuscate(domainName), qType)
 
 	req := newRequest(dns.Fqdn(domainName), qType)
+
 	response, err := r.next.Resolve(ctx, req)
+	if err != nil {
+		util.LogOnError(ctx, fmt.Sprintf("can't prefetch '%s' ", domainName), err)
 
-	if err == nil {
-		if response.Res.Rcode == dns.RcodeSuccess {
-			packed, err := response.Res.Pack()
-			if err != nil {
-				logger.Error("unable to pack response", err)
+		return nil, 0
+	}
 
-				return nil, 0
-			}
+	if response.Res.Rcode == dns.RcodeSuccess {
+		respCopy := response.Res.Copy()
 
-			return &packed, r.adjustTTLs(response.Res.Answer)
+		// don't cache any EDNS OPT records
+		util.RemoveEdns0Record(respCopy)
```

Comment:
Nice catch!

Based on my other comments and this being missing before, I think we should try and factorize the code to have a single func we can call from here and `putInCache`. Maybe something that adds the response to Redis, and returns the local cache TTL + the entry with adjusted TTL.